### PR TITLE
Make the Enter activity use the closest target position.

### LINKED ISSUE #8064: Remote infantry crushing
Notice how the engineer passes trough the topmost tiberium tile and dies when the harvester drives on the same tile later. Yes and this affects commandoes too. They can be crushed long after they have entered a building by driving a vehicle over the tile they entered from thus saving said building. No stealth tanks around because spectators see them and it is only 5 mins into the game:
crush
I just wanted to put this gif here thats all :P
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/12535347-remote-infantry-crushing?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #11416: TD/RA Tanya/Commando/Engineer Building entry.
Is there a way for the units Tanya, Commando, Engineer (TD) to have them enter into only part of the building instead of the center? This would help prevent the units being crushed which are not standing on the tile as they do the entry animations.

### PR DESCRIPTION
**Pull request by pchote**
Closes #8064.
Closes #11416.

### CODE REVIEW

File: OpenRA.Mods.Common/Activities/Enter.cs [hunk 1]
```diff
@@ -145,7 +146,10 @@ ReserveStatus TryReserveElseTryAlternateReserve(Actor self)
 						Target t = target;
 						if (!TryGetAlternateTarget(self, tries, ref t))
 							return ReserveStatus.TooFar;
-						if ((target.CenterPosition - self.CenterPosition).HorizontalLengthSquared <= (t.CenterPosition - self.CenterPosition).HorizontalLengthSquared)
+
+						var targetPosition = target.Positions.PositionClosestTo(self.CenterPosition);
+						var alternatePosition = t.Positions.PositionClosestTo(self.CenterPosition);
```
Comment (rob-v) [inline OpenRA.Mods.Common/Activities/Enter.cs]: Shouldn't we use here bool targetCenter to decide the target pos?
or in this code 
`inner = move.MoveToTarget(self, targetCenter ? Target.FromPos(target.CenterPosition) : target); // Approach`
also use PositionClosestTo? Though for ReserveStatus.TooFar it probably doesn't matter too much.

Comment (pchote) [inline OpenRA.Mods.Common/Activities/Enter.cs]: Actually, it turns out that `targetCenter` has nothing to do with which position to target.  It is used to suppress repathing during group-transport orders.

I added a commit to replace that variable to make this clearer.

**Comment by pchote** [comment]
Updated.

**Review by rob-v — APPROVED** [review]
👍

**Review by obrakmann — APPROVED** [review]
👍

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/8eb834a69c363b81a12ac0e96c4d39c48edc8c46)

**Pull request merged**